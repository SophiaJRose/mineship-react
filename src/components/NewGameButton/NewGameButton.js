import * as React from "react"

function NewGameButton(props) {
    return (
        <button onClick={props.onClick}>New Game</button>
    )
}

export { NewGameButton }