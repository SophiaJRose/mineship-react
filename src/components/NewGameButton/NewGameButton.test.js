import React from "react"
import { render, fireEvent, screen } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import { NewGameButton } from "./NewGameButton"

describe("<NewGameButton /> tests", () => {
    it("should render as default button with text New Game", () => {
        const { container } = render(<NewGameButton />)
        expect(container.firstChild).toMatchSnapshot()
        expect(screen.getByRole("button", {name: /new game/i})).toBeInTheDocument()
    })

	it("should call the click callback handler", () => {
		// Arrange
		const onClick = jest.fn()
		const { container } = render(<NewGameButton onClick={onClick} />)
		// Act
		fireEvent.click(container.firstChild)
		// Assert
		expect(onClick).toHaveBeenCalledTimes(1)
	})
})