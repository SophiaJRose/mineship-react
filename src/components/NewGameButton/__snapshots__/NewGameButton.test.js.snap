// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewGameButton /> tests should render as default button with text New Game 1`] = `
<button>
  New Game
</button>
`;
